### DISCUSSION TITLE
An ability to upload text documents into a chat.

### DISCUSSION THREAD

**Discussion by juicesharp** [danny-avila/LibreChat#1496] [Feature Requests & Suggestions]
ChatGPT allows to upload text files into a chat. LibreChat allows to upload only image formats. Can you extend support of other formats beyond the pictures?
![2024-01-05_13-59-08](https://github.com/danny-avila/LibreChat/assets/7575510/b13d299c-dcf9-4f0e-85cf-7b7a83208010)

**Comment by danny-avila** [comment]
It’s currently in development, hoping to be done by end of month.

**Reply by juicesharp** [reply]
Thank you for the quick response.

**Comment by mk11232** [comment]
Any update to this feature. It's a basic feature that is missing which makes this tool not as useful as many other ones out there :(

**Reply by danny-avila** [reply]
Yup. I'm looking to roll out file support gradually. First with Assistants API, then native RAG/file import solutions.

A preview should be out by end of day for you to try, but it will not officially merge to main branch until end of week.

![image](https://github.com/danny-avila/LibreChat/assets/110412045/003f4f1c-57f2-44c6-8920-9f595ddcef93)

**Reply by sweitzel** [reply]
@danny-avila  I have a existing OpenSearch cluster with all my knowledge base data in it and with proper embeddings. So basically I want that LibreChat accesses the data from OpenSearch (e.g. using [Hybrid Search](https://opensearch.org/docs/latest/search-plugins/hybrid-search/)). Can I use this already today together with LibreChat - maybe by creating a Plugin?
Or will this supported by the RAG feature mentioned on the road map?

**Moved** to separate discussion: https://github.com/danny-avila/LibreChat/discussions/2178

**Comment by danny-avila** [comment]
FYI general file support is now implemented and live on the repo. Still need to write documentation but the RAG API that the project is using I made is here: https://github.com/danny-avila/rag_api
